curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -537,95 +530,138 @@ bool FileCacheManager::ReadKVRequestFromS3(const std::string &name,
     if (*ret < 0) {
         LOG(ERROR) << "object " << name << " read from s3 fail, ret = " << *ret;
         return false;
-    } else if (s3ClientAdaptor_->s3Metric_.get()) {
+    }
+
+    if (s3ClientAdaptor_->s3Metric_) {
         s3ClientAdaptor_->CollectMetrics(
             &s3ClientAdaptor_->s3Metric_->adaptorReadS3, length, start);
     }
 
     return true;
 }
 
-int FileCacheManager::ReadKVRequest(
+FileCacheManager::ReadStatus FileCacheManager::ReadKVRequest(
     const std::vector<S3ReadRequest> &kvRequests, char *dataBuf,
     uint64_t fileLen) {
+    readTaskPool_.Start(readCacheThreads_);
```

Comment:
maybe the thread pool shouldn't be in `FileCacheManager`, If a large number of files are read at the same time, the number of threads can get out of hand?

Reply:
Maybe bthread can work? Bthread is an M:N thread, multiple Bthreads can be declared in the same Read thread

Reply:
> Maybe bthread can work? Bthread is an M:N thread, multiple Bthreads can be declared in the same Read thread
sorry, I don't think it's a good way, we just need a configurable number of threads, so i think move the thread pool to `FsCacheManager` maybe better?

Reply:
Do you mean that all `FileCacheManager` share a thread pool?

Reply:
> Do you mean that all `FileCacheManager` share a thread pool?

yes

Reply:
I'll try it :)